src/ClientServices/ProcessClient.cpp

```diff
@@ -94,15 +95,24 @@ ErrorMessageOr<std::string> ProcessClient::FindDebugInfoFile(
   *request.mutable_additional_search_directories() = {additional_search_directories.begin(),
                                                       additional_search_directories.end()};
 
-  std::unique_ptr<grpc::ClientContext> context = CreateContext();
+  const std::unique_ptr<grpc::ClientContext> context = CreateContext();
 
-  grpc::Status status = process_service_->GetDebugInfoFile(context.get(), request, &response);
-  if (!status.ok()) {
+  const grpc::Status status = process_service_->GetDebugInfoFile(context.get(), request, &response);
+
+  if (status.ok()) {
+    return std::filesystem::path{response.debug_info_file_path()};
+  }
+
+  if (status.error_code() == grpc::UNKNOWN) {
     ORBIT_ERROR("gRPC call to GetDebugInfoFile failed: %s", status.error_message());
     return ErrorMessage(status.error_message());
   }
 
-  return response.debug_info_file_path();
+  if (status.error_code() == grpc::NOT_FOUND) {
+    return orbit_base::NotFound{status.error_message()};
+  }
+
+  ORBIT_UNREACHABLE();
```

Comment:
Another option here would be to return `ErrorMessage` in all cases except `ok` and `NOT_FOUND`. This way you don't use the `ORBIT_UNREACHABLE` to make assumptions on what exact statuses the service can send.

And another one would be to return a different `ErrorMessage` instead of `ORBIT_UNREACHABLE`.

I know there are different ways of looking at how we use checks that abort so I don't have a strong opinion. I'm just thinking about the case where we possibly add a rare status returned by the service: if we miss to consider it here we crash in that rare case instead of simply having an `ErrorMessage` like in the `UNKNOWN` case.

What's your take on this?

Reply:
This is a very good point you are raising. Maybe gRPC also returns a different status code when the connection breaks down, or there is a unmarshalling error, etc. 

I removed the UNREACHABLE and always return an ErrorMessage now. I still discern between just forwarding the error message in the `UNKNOWN` status code case. For any other status code there is a more detailed error message. I also added explanatory comments

Reply:
Great.